app/services/key_rotator/hmac_fingerprinter.rb

```diff
@@ -14,7 +14,7 @@ def rotate(user:, pii_attributes: nil, profile: nil)
 
     def rotate_email_fingerprint(user)
       ee = EncryptedAttribute.new_from_decrypted(user.email)
-      user.update_columns(email_fingerprint: ee.fingerprint)
+      UpdateUser.new(user: user, attributes: { email_fingerprint: ee.fingerprint }).call
```

Comment:
Given that my decision to use `update_columns` was both for performance reasons (rotating millions of records can take time), and to avoid the `updated_at` timestamp, I would prefer to see it stay in this instance. ActiveRecord has multiple ways to do things, because a single method does not fit every need. I would rather keep it here. Adding more indirection doesn't help here IMO.